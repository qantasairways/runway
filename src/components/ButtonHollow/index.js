import React from 'react';
import ButtonBase, {
  buttonPropTypes,
  buttonDefaultProps
} from '../../baseComponents/Button';
import { CSS_SELECTOR_ACTIVE, CSS_SELECTOR_HOVER } from '../../constants/css';

const themeStyles = {
  button: theme => ({
    backgroundColor: theme.colours.transparent,
    boxShadow: `0 0 0 2px ${theme.colours.primary} inset`,
    color: theme.colours.primary,
    fontWeight: 'normal',
    [CSS_SELECTOR_HOVER]: {
      boxShadow: `0 0 0 2px ${theme.colours.primaryDark} inset`,
      color: theme.colours.primaryDark
    },
    [CSS_SELECTOR_ACTIVE]: {
      boxShadow: `0 0 0 2px ${theme.colours.primaryDark} inset`,
      color: theme.colours.primaryDark
    }
  })
};

const ButtonHollow = props => (
  <ButtonBase {...props} themeStyles={themeStyles} />
);

ButtonHollow.propTypes = {
  ...buttonPropTypes
};

ButtonHollow.defaultProps = {
  ...buttonDefaultProps
};

export default ButtonHollow;
