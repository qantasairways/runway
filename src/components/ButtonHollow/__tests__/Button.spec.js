import React from 'react';
import ButtonHollow from '..';
import { shallow } from 'enzyme';

describe('ButtonHollow', () => {
  it('should render correctly with defaults', () => {
    const component = shallow(<ButtonHollow />);

    expect(component).toMatchSnapshot();
  });

  it('should render correctly with props', () => {
    const component = shallow(
      <ButtonHollow type="submit" onClick={() => 1}>
        button text
      </ButtonHollow>
    );

    expect(component).toMatchSnapshot();
  });
});
