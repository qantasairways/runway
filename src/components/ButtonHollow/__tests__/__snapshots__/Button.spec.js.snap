// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonHollow should render correctly with defaults 1`] = `
<WithTheme(ThemedButton)
  disabled={false}
  inline={false}
  onClick={[Function]}
  themeStyles={
    Object {
      "button": [Function],
    }
  }
  type="button"
>
  button
</WithTheme(ThemedButton)>
`;

exports[`ButtonHollow should render correctly with props 1`] = `
<WithTheme(ThemedButton)
  disabled={false}
  inline={false}
  onClick={[Function]}
  themeStyles={
    Object {
      "button": [Function],
    }
  }
  type="submit"
>
  button text
</WithTheme(ThemedButton)>
`;
